### PR TITLE
Update Docker base image to python3.12-slim (#3458)

### PR DESCRIPTION
**Pull request by PikaBlue107**
## Overview: What does this pull request change?
Simply updating the Docker base image from python 3.8 -> 3.12.

This lets users of the Docker distribution of `manim` take advantage of lots of new Python features (especially type hints).

This should not cause any breaking changes AFAIK.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by naveen521kk** [comment]
Did you build and try manim with these changes?

**Review by behackl — CHANGES_REQUESTED** [review]
It seems like `python:3.12-slim` might be a bit too recent: when I try to build the image locally, I get

```py
#11 155.1       zmq/backend/cython/_device.c:5408:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
#11 155.1        5408 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
#11 155.1             |                                                     ^~
#11 155.1       zmq/backend/cython/_device.c:5463:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
#11 155.1        5463 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
#11 155.1             |                                                     ^~
#11 155.1       zmq/backend/cython/_device.c: In function ‘__Pyx_PyInt_As_long’:
#11 155.1       zmq/backend/cython/_device.c:5680:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
#11 155.1        5680 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
#11 155.1             |                                                     ^~
#11 155.1       zmq/backend/cython/_device.c:5735:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
#11 155.1        5735 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
#11 155.1             |                                                     ^~
#11 155.1       zmq/backend/cython/_device.c: In function ‘__Pyx_PyIndex_AsSsize_t’:
#11 155.1       zmq/backend/cython/_device.c:6086:45: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
#11 155.1        6086 |     const digit* digits = ((PyLongObject*)b)->ob_digit;
#11 155.1             |                                             ^~
#11 155.1       ************************************************
#11 155.1       ************************************************
#11 155.1       ************************************************
#11 155.1       error: command '/usr/bin/gcc' failed with exit code 1
#11 155.1       [end of output]
#11 155.1
#11 155.1   note: This error originates from a subprocess, and is likely not a problem with pip.
#11 155.1   ERROR: Failed building wheel for pyzmq
```

Perhaps we should go for 3.11 instead?

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.8-slim
+FROM python:3.12-slim
```
Comment (behackl) [inline docker/Dockerfile:1]:
```suggestion
FROM python:3.11-slim
```

**Review by behackl — APPROVED** [review]
Builds successfully locally now, we'll see how this combines with the modified version requirements from the other PR. Thanks!

**Comment by PikaBlue107** [comment]
Thank you for progressing the PR! I've been trying and failing to find the time myself. Cheers!

**Pull request merged**